packages/jupyter-widgets/src/manager/widgetLoader.ts

```diff
@@ -0,0 +1,89 @@
+import * as base from "@jupyter-widgets/base";
+import * as controls from "@jupyter-widgets/controls";
+
+let cdn = "https://unpkg.com";
+
+/**
+ * Constructs a well formed module URL for requireJS 
+ * mapping the modulename and version from the base CDN URL
+ * @param moduleName Name of the module corresponding to the widget package
+ * @param moduleVersion Module version returned from kernel
+ */
+function moduleNameToCDNUrl(moduleName: string, moduleVersion: string): string {
+  let packageName = moduleName;
+  let fileName = "index.js"; // default filename
+  // if a '/' is present, like 'foo/bar', packageName is changed to 'foo', and path to 'bar'
+  // We first find the first '/'
+  let index = moduleName.indexOf('/');
+  if (index !== -1 && moduleName[0] === '@') {
+    // if we have a namespace, it's a different story
+    // @foo/bar/baz should translate to @foo/bar and baz
+    // so we find the 2nd '/'
+    index = moduleName.indexOf('/', index + 1);
+  }
+  if (index !== -1) {
+    fileName = moduleName.substr(index + 1);
+    packageName = moduleName.substr(0, index);
+  }
+  let moduleNameString = packageName;
+  if(moduleVersion){
+    moduleNameString = `${packageName}@${moduleVersion}`;
+  }
+  return `${cdn}/${moduleNameString}/dist/${fileName}`;
+}
+
+/**
+ * Initialize dependencies that need to be preconfigured for requireJS module loading
+ * Here, we define the jupyter-base, controls package that most 3rd party widgets depend on
+ * We also override the cdn end point (if applicable)
+ */
+export function initRequireDeps(){
+  // Export the following for `requirejs`.
+  // tslint:disable-next-line: no-any no-function-expression no-empty
+  const define = (window as any).define || function () {};
+  define("@jupyter-widgets/controls", () => controls);
+  define("@jupyter-widgets/base", () => base);
+}
+
+/**
+ * Overrides the default CDN base URL by querying the DOM for script tags
+ * We follow the same pattern as defined in HTML manager class of ipywidgets
+ * https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/html-manager/src/libembed-amd.ts
+ */
+export function overrideCDNBaseURL(){
+  // find the data-cdn for any script tag, assuming it is only used for embed-amd.js
+  const scripts = document.getElementsByTagName("script");
+  Array.prototype.forEach.call(scripts, (script: HTMLScriptElement) => {
+    cdn = script.getAttribute("data-jupyter-widgets-cdn") || cdn;
```

Comment:
Let's add this to documentation somewhere so people know how they can override.

Reply:
I have updated the documentation of this method. I've made a note of this point. I am planning to add more detailed documentation and update the `README` after getting a little more feedback from the nteract maintainers when we submit this PR to nteract.